consul/src/main/java/com/linecorp/armeria/internal/consul/HealthClient.java

```diff
@@ -0,0 +1,175 @@
+ * <a href="https://www.consul.io/api/health.html">Health Endpoint API</a>.
+ */
+final class HealthClient {
+
+    private static final Logger logger = LoggerFactory.getLogger(HealthClient.class);
+
+    private static final String PASSING_PARAM = "?passing=true";
+
+    static HealthClient of(ConsulClient consulClient) {
+        return new HealthClient(consulClient);
+    }
+
+    private final WebClient client;
+    private final ObjectMapper mapper;
+
+    private HealthClient(ConsulClient client) {
+        this.client = client.consulWebClient();
+        mapper = client.getObjectMapper();
+    }
+
+    /**
+     * Returns a healthy endpoint list with service name.
+     */
+    CompletableFuture<List<Endpoint>> healthyEndpoints(String serviceName) {
+        requireNonNull(serviceName, "serviceName");
+        final StringBuilder path = new StringBuilder("/health/service/");
+        PercentEncoder.encodeComponent(path, serviceName);
+        path.append(PASSING_PARAM);
+        return client
+                .get(path.toString())
+                .aggregate()
+                .handle((response, cause) -> {
+                    if (cause != null) {
+                        logger.warn("Unexpected exception while fetching the registry from Consul: {}" +
+                                    " (serviceName: {})", client.uri(), serviceName,
+                                    cause);
+                        return null;
+                    }
+
+                    final HttpStatus status = response.status();
+                    final String content = response.contentUtf8();
+                    if (!status.isSuccess()) {
+                        logger.warn("Unexpected response from Consul: {} (status: {}, content: {}," +
+                                    " serviceName: {})", client.uri(), status,
+                                    content, serviceName);
+                        return null;
+                    }
+
+                    try {
+                        return Arrays.stream(mapper.readValue(content, HealthService[].class))
+                                     .map(HealthClient::toEndpoint)
+                                     .collect(toImmutableList());
+                    } catch (IOException e) {
+                        logger.warn("Unexpected exception while parsing a response from Consul: {}" +
+                                    " (content: {}, serviceName: {})",
+                                    client.uri(), content, serviceName, e);
+                        return null;
+                    }
+                });
+    }
+
+    private static Endpoint toEndpoint(HealthService healthService) {
+        final String host;
+        if (!healthService.service.address.isEmpty()) {
+            host = healthService.service.address;
+        } else if (!healthService.node.address.isEmpty()) {
+            host = healthService.node.address;
+        } else {
+            host = "127.0.0.1";
```

Comment:
If `serviceAddress` and `address` are both `null`, I think we should return `null` (don't forget to mark `@Nullable`) and filter where this is called.
`.filter(Objects::nonNull)`

Reply:
@minwoox , how about using `Optional`?

Reply:
Well, we prefer to use `@Nullable` instead of `Optional` for the API. So let's use just `@Nullable` for consistency. 🙇

Reply:
Yes, I got it.